featurebyte/worker/task/catalog_cleanup.py

```diff
@@ -214,44 +286,24 @@ async def _cleanup_catalog(
                 percent=int(100 * i / len(self.catalog_specific_model_class_pairs)),
                 message="Cleaning up Mongo records",
             )
-            await self.persistent.delete_many(
+            with_audit = True
+            model_settings = getattr(model_class, "Settings", None)
+            if model_settings:
+                with_audit = getattr(model_settings, "auditable", True)
```

Comment:
Is there a reason to use `getattr` instead of directly accessing the attributes? If the auditable attribute doesn't exist, is it intended to assume it is auditable?

Reply:
For certain class like table model, we don't pass class object. We only pass the function used to construct the object. In this case, the object does not have `Settings` class.